pkg/search/postgres/query/common.go

```diff
@@ -91,5 +95,26 @@ func MatchFieldQuery(dbField *walker.Field, value string, highlight bool, now ti
 	if !ok {
 		return nil, nil
 	}
-	return matchFieldQuery(dbField, field, value, highlight, now)
+
+	qualifiedColName := dbField.Schema.Table + "." + dbField.ColumnName
+	dataType := dbField.DataType
+	var goesIntoHavingClause bool
+	if derivedMetadata != nil {
+		switch derivedMetadata.DerivationType {
+		case search.CountDerivationType:
+			qualifiedColName = fmt.Sprintf("count(%s)", qualifiedColName)
+			dataType = walker.Integer
+			goesIntoHavingClause = true
```

Comment:
All derivation types can go into having because having is where+aggregations.

Reply:
Hmm, I'm not so sure that's true. We can figure that out when we add support for other derivation types, I guess?